src/Domain/Creators/TimeDependence/UniformRotationAboutZAxis.hpp

```diff
@@ -0,0 +1,158 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <limits>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
+#include "Domain/CoordinateMaps/Identity.hpp"
+#include "Domain/CoordinateMaps/TimeDependent/Rotation.hpp"
+#include "Domain/Creators/TimeDependence/GenerateCoordinateMap.hpp"
+#include "Domain/Creators/TimeDependence/TimeDependence.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/TMPL.hpp"
+
+/// \cond
+namespace domain {
+namespace FunctionsOfTime {
+class FunctionOfTime;
+}  // namespace FunctionsOfTime
+namespace CoordinateMaps {
+namespace TimeDependent {
+template <typename Map1, typename Map2>
+class ProductOf2Maps;
+}  // namespace TimeDependent
+}  // namespace CoordinateMaps
+
+template <typename SourceFrame, typename TargetFrame, typename... Maps>
+class CoordinateMap;
+}  // namespace domain
+
+namespace Frame {
+struct Grid;
+struct Inertial;
+}  // namespace Frame
+/// \endcond
+
+namespace domain {
+namespace creators {
+namespace time_dependence {
+/*!
+ * \brief A uniform rotation about the \f$z\f$ axis:
+ * \f{eqnarray*}
+ * x &\to& x \cos \alpha(t) - y \sin \alpha(t) \\
+ * y &=\to& y \sin \alpha(t) + y \cos \alpha(t)
+ * \f},
+ * where \f$\alpha(t)\f$ is a `FunctionOfTime`. For 3 spatial dimensions,
+ * \f$z \to z\f$, and the rotation is implemented as a product of the 2D
+ * rotation and an identity map. The rotation is undefined (and therefore
+ * unimplemented here) for 1 spatial dimension.
+ */
+template <size_t MeshDim>
```

Comment:
Add a static assert that `MeshDim > 1`? Then you can remove the conditionals for `MeshDim == 1`. Else, mention in the docs above that this is the identity in 1D.

Reply:
Good catch! @nilsdeppe and I already decided to make 1D undefined, but I missed removing the if dim == 1 logic here when I made that change.

But why not use a Requires<> in the template parameter instead of a static_assert?

Reply:
static assert would give a much better error message